root/programs/C#/Samples/WebApp_sample/MVC_Sample/MVC_Sample/Controllers/HomeController.cs

```diff
@@ -115,5 +182,95 @@ public ActionResult Logout()
             FormsAuthentication.SignOut();
             return this.Redirect(Url.Action("Index", "Home"));
         }
+
+        /// <summary>OAuthAuthorizationCodeGrantClient</summary>
+        /// <param name="code">string</param>
+        /// <param name="state">string</param>
+        /// <returns>ActionResultを非同期的に返す</returns>
+        [HttpGet]
+        [AllowAnonymous]
+        public async Task<ActionResult> OAuthAuthorizationCodeGrantClient(string code, string state)
+        {
+            if (state == this.State) // CSRF(XSRF)対策のstateの検証は重要
+            {
+                HttpClient httpClient = new HttpClient();
+
+                HttpRequestMessage httpRequestMessage = null;
+                HttpResponseMessage httpResponseMessage = null;
+
+                // HttpRequestMessage (Method & RequestUri)
+                httpRequestMessage = new HttpRequestMessage
+                {
+                    Method = HttpMethod.Post,
+                    RequestUri = new Uri("http://localhost:63359/MultiPurposeAuthSite/OAuthBearerToken"),
+                };
+
+                // HttpRequestMessage (Headers & Content)
+                httpRequestMessage.Headers.Authorization = new AuthenticationHeaderValue(
+                    "Basic",
+                    Convert.ToBase64String(System.Text.Encoding.ASCII.GetBytes(
+                        string.Format("{0}:{1}",
+                            "f53469c17c5a432f86ce563b7805ab89",
+                            "cKdwJb6mRKVIJpGxEWjIC94zquQltw_ECfO-55p21YM"))));
+
+                httpRequestMessage.Content = new FormUrlEncodedContent(
+                    new Dictionary<string, string>
+                    {
+                        { "grant_type", "authorization_code" },
+                        { "code", code },
+                        { "redirect_uri", System.Web.HttpUtility.HtmlEncode("http://localhost:58496/MVC_Sample/Home/OAuthAuthorizationCodeGrantClient") },
+                    });
+
+                // HttpResponseMessage
+                httpResponseMessage = await httpClient.SendAsync(httpRequestMessage);
+                string response = await httpResponseMessage.Content.ReadAsStringAsync();
+
+                // 汎用認証サイトはOIDCをサポートしたのでid_tokenを取得し、検証可能。
+                Base64UrlTextEncoder base64UrlEncoder = new Base64UrlTextEncoder();
+                Dictionary<string, string> dic = JsonConvert.DeserializeObject<Dictionary<string, string>>(response);
+
+                // id_tokenの検証コード
+                string id_token = dic["id_token"];
+
+                JWT_RS256 jwtRS256 = new JWT_RS256(
+                    @"C:\Git1\MultiPurposeAuthSite\root\programs\MultiPurposeAuthSite\CreateClientsIdentity\CreateClientsIdentity_RS256.cer", "test");
```

Comment:
@daisukenishino 

It might be a good idea to specify this path in `app.config`.